salesagility/SuiteCRM-Core
# Feature date components change proposal

In the process of solving https://github.com/salesagility/SuiteCRM-Core/issues/399 and understanding the Date component in edit mode and filter mode, to improve the readability and functionality this proposal was created.

## Description




  - The real fix was on `core/app/core/src/lib/fields/base/base-field.component.ts:257` of fix https://github.com/salesagility/SuiteCRM-Core/issues/323, but I do not want to leave this change proposal uncommitted.
  - `core/app/core/src/lib/fields/date/templates/edit/date.component.ts:[64~75, 97~124]`: three new functions were created to centralise value assignation
  - `core/app/core/src/lib/fields/date/templates/filter/date.component.ts:[67~74, 101~127]`: four new functions were created to centralise value assignation.
  - `core/app/core/src/lib/fields/date/templates/edit/date.component.html:[35~36]` and `core/app/core/src/lib/fields/date/templates/filter/date.component.html:[35~36]` we do not need to react to `ngModelChange` of the component to update `this.field.value` and `this.dateModel`, that is done through `this.field.formControl` changes that triggers `setFieldValue` function.

## Motivation and Context

I do not want to leave this change proposal uncommitted.

## How To Test This

1. Open Opportunities in ListView
2. Click the Filter button to open the Filter
3. Check the Expected Close Date field and try to filter by.

## Types of changes


### Final checklist
